asciidoctorj-api/src/main/java/org/asciidoctor/ast/Catalog.java

```diff
@@ -0,0 +1,32 @@
+package org.asciidoctor.ast;
+
+import java.util.List;
+import java.util.Map;
+
+
+public interface Catalog {
+
+    /**
+     * Note that footnotes are only available after `Document.getContent()` has been called.
+     *
+     * A converter uses cataloged footnotes to render them, presumably, at the bottom of a document.
+     *
+     * @return footnotes occurring in document.
+     */
+    List<Footnote> getFootnotes();
+
+    /**
+     * Refs is a map of asciidoctor ids to asciidoctor document elements.
+     *
+     * For example, by default, each section is automatically assigned an id.
+     * In this case the id would map to a {@link Section} element.
+     *
+     * Ids can also be explicitly assigned by document authors to any document element.
+     * See https://asciidoctor.org/docs/user-manual/#id
+     *
+     * A converter might use cataloged refs to lookup ids to support rendering inline anchors.
+     *
+     * @return a map of ids to elements that asciidoctor has collected from the document.
+     */
+    Map<String, Object> getRefs();
```

Comment:
Could this be a `Map<String, ContentNode>`?
Or is it possible that this contains sth else than an AST node?

Reply:
That is a good question that I don't know the answer to.  

In my experiments and testing I did not see a case where a ref id points to anything but a `ContentNode`, but I do see allowances for it possibly being something else in the [html5 Ruby converter](https://github.com/asciidoctor/asciidoctor/blob/4f155298308b646f3327619f6b0dfdbe52124366/lib/asciidoctor/converter/html5.rb#L1115), so I cautiously left it as an `Object`.

Reply:
I see, then it is probably safer to keep it as Object for now.

Reply:
I think so too.

Reply:
Checked the code in Asciidoctor core again. I agree, if Asciidoctor core is not sure whether it is an AbstractNode or not, we shouldn't be either.